### PR TITLE
Feature/17 UI improvements

### PR DESCRIPTION
**Pull request by s6thgehr**

### CODE REVIEW

File: packages/app/src/components/Button.tsx [hunk 1]
```diff
@@ -2,24 +2,25 @@ import clx from "classnames";
 import React, { ReactNode } from "react";
 
 interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
-  variant?: "primary" | "secondary";
+  variant?: "green" | "red" | "unactive";
```
Comment (ronnyhaase) [inline packages/app/src/components/Button.tsx]: I'd rather stay with states than colors. How about primary, secondary and danger (for red)

File: packages/app/src/components/Button.tsx [hunk 2]
```diff
@@ -2,24 +2,25 @@ import clx from "classnames";
...
 }) => {
   return (
     <button
       className={clx(
         "px-8 py-4 rounded-lg text-white text-xl",
         {
-          "bg-green": variant === "primary",
-          "bg-outset": variant === "secondary",
+          "bg-green": variant === "green",
+          "bg-red-400": variant === "red",
```
Comment (ronnyhaase) [inline packages/app/src/components/Button.tsx]: Please add a color to tailwind.config.js named "danger" and use "bg-danger" here

File: packages/app/src/components/StakeForm.tsx [hunk 3]
```diff
@@ -11,44 +12,87 @@ import { toFixedWithPrecision, toSol } from "../lib/util";
...
-}) => (
-  <div className={className}>
-    <div className="p-8 pt-3 bg-background">
-      <div className="text-right">
-        Balance:{" "}
-        <span className="text-blue">
-          {balance ? toFixedWithPrecision(toSol(balance)) : "-"} SOL
-        </span>
+}) => {
+  const [isMax, setIsMax] = useState(false);
```
Comment (ronnyhaase) [inline packages/app/src/components/StakeForm.tsx]: We don't need state here, on Max click just setAmount(balance)
Keeping track if amount == balance just adds complexity with no real user benefit

File: packages/app/src/components/StakeForm.tsx [hunk 4]
```diff
@@ -11,44 +12,87 @@ import { toFixedWithPrecision, toSol } from "../lib/util";
...
+          <span className="flex flex-row mt-2">
+            <button
+              className={clx(
+                "text-xl my-auto mx-4 px-4 py-1 border-solid border-[1px] border-green rounded-md bg-green",
+                { "bg-opacity-25": !isMax }
+              )}
+              onClick={() => {
+                setIsMax((prevState) => {
+                  if (prevState) {
+                    // setAmount here creates a warning but I don't understand why, it says this is a bad setState call
```
Comment (ronnyhaase) [inline packages/app/src/components/StakeForm.tsx]: When we get rid of that state all problems solved :)

File: packages/app/src/components/StakeForm.tsx [hunk 5]
```diff
@@ -11,44 +12,87 @@ import { toFixedWithPrecision, toSol } from "../lib/util";
...
+            >
+              Max
+            </button>
+
+            <input
+              className="w-28 border-none bg-transparent text-3xl text-right"
+              type="number"
+              min="0"
+              // It still show comma instead of point for decimal
+              lang="en"
```
Comment (ronnyhaase) [inline packages/app/src/components/StakeForm.tsx]: Remove it, the browser handles localization ;).

File: packages/app/src/pages/StakeDashboard.tsx [hunk 6]
```diff
@@ -114,6 +126,7 @@ export const StakeDashboard: FC = () => {
           </div>
         )}
         <StakeForm
```
Comment (ronnyhaase) [inline packages/app/src/pages/StakeDashboard.tsx]: Can you refactor it to have a StakeForm and a UnstakeForm which are shown depending on isStakeSelected, this way we can clearly separate the business logic for staking and unstaking and don't have add logic in event handlers if it's this or that state.

**Review by ronnyhaase — CHANGES_REQUESTED** [review]
Great job! Just some minor things and one small refactoring I'd suggest. Happy to merge afterwards!

File: packages/app/src/components/StakeForm.tsx [hunk 7]
```diff
@@ -11,44 +12,87 @@ import { toFixedWithPrecision, toSol } from "../lib/util";
...
+          <div className="text-right">
+            Balance:{" "}
+            <span className="text-blue">
+              {balance ? toFixedWithPrecision(toSol(balance)) : "-"} SOL
+            </span>
+          </div>
+          <span className="flex flex-row mt-2">
+            <button
+              className={clx(
+                "text-xl my-auto mx-4 px-4 py-1 border-solid border-[1px] border-green rounded-md bg-green",
```
Comment (ronnyhaase) [inline packages/app/src/components/StakeForm.tsx]: Use border-1 for border-[1px] I think it is 1px

File: packages/app/src/components/StakeForm.tsx [hunk 8]
```diff
@@ -11,44 +12,87 @@ import { toFixedWithPrecision, toSol } from "../lib/util";
...
+        <img src="solana-logo.png" className="h-24 my-auto" />
+        <div className="my-auto">
+          <div className="text-right">
+            Balance:{" "}
+            <span className="text-blue">
+              {balance ? toFixedWithPrecision(toSol(balance)) : "-"} SOL
+            </span>
+          </div>
+          <span className="flex flex-row mt-2">
+            <button
```
Comment (ronnyhaase) [inline packages/app/src/components/StakeForm.tsx]: As discussed in Telegram, let's get rid of the max button and on click on balance set to max.

File: packages/app/src/components/Button.tsx (hunk 1 above)
Comment (s6thgehr) [reply packages/app/src/components/Button.tsx]: Done

File: packages/app/src/components/Button.tsx (hunk 2 above)
Comment (s6thgehr) [reply packages/app/src/components/Button.tsx]: Done.

File: packages/app/src/components/StakeForm.tsx (hunk 5 above)
Comment (s6thgehr) [reply packages/app/src/components/StakeForm.tsx]: Done

File: packages/app/src/components/StakeForm.tsx (hunk 8 above)
Comment (s6thgehr) [reply packages/app/src/components/StakeForm.tsx]: Done.

File: packages/app/src/components/StakeForm.tsx (hunk 4 above)
Comment (s6thgehr) [reply packages/app/src/components/StakeForm.tsx]: Done.

File: packages/app/src/components/StakeForm.tsx (hunk 3 above)
Comment (s6thgehr) [reply packages/app/src/components/StakeForm.tsx]: Done.

File: packages/app/src/components/StakeForm.tsx (hunk 7 above)
Comment (s6thgehr) [reply packages/app/src/components/StakeForm.tsx]: I think-border-1 does not exist, only even border widths. But the button is removed anyway.

File: packages/app/src/pages/StakeDashboard.tsx (hunk 6 above)
Comment (s6thgehr) [reply packages/app/src/pages/StakeDashboard.tsx]: Done.

File: packages/app/src/components/StakeForm.tsx [hunk 9]
```diff
@@ -1,59 +1,21 @@
...
-      <input
-        className="w-full border-none bg-transparent text-3xl text-right"
-        type="number"
-        placeholder="0.00"
-        onChange={(ev) => setAmount(ev.currentTarget.value)}
-      />
-    </div>
-  </div>
-);
+import AmountInput from "./AmountInput";
```
Comment (ronnyhaase) [inline packages/app/src/components/StakeForm.tsx:8]: Good catch, makes sense! ❤️

**Review by ronnyhaase — APPROVED** [review]

**Pull request merged**